File: leet/array_string/greatest_number_of_candies_1431.py
from typing import List

class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        max_c = max(candies)
        return [True if n+extraCandies >= max_c else False for n in candies]
    
    # def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
    #     tf = []
    #     max_c = max(candies)
    #     for n in candies:
    #         if n+extraCandies >= max_c:
    #             tf.append(True)
    #         else:
    #             tf.append(False)

    #     return tf


ss = Solution()
ss.kidsWithCandies(candies = [2,3,5,1,3], extraCandies = 3)

File: leetcode_75/array_string/greatest_number_of_candies_1431.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> ll = new ArrayList<>(candies.length);
        int max = Arrays.stream(candies).max().orElseThrow();
        for (int i : candies) {
            if (i + extraCandies >= max) {
                ll.add(true);
            } else {
                ll.add(false);
            }
        }
        return ll;
    }
}

// 7kyu_high_and_low
// 可以參考